src/WebJobs.Script.WebHost/Filters/AuthorizationLevelAttribute.cs

```diff
@@ -33,12 +33,20 @@ public AuthorizationLevelAttribute(AuthorizationLevel level)
                 throw new ArgumentNullException("actionContext");
             }
 
-            // determine the authorization level for the function and set it
-            // as a request property
-            var secretManager = actionContext.ControllerContext.Configuration.DependencyResolver.GetService<ISecretManager>();
+            AuthorizationLevel requestAuthorizationLevel = actionContext.Request.GetAuthorizationLevel();
+            
+            // If the request has not yet been authenticated, authenticate it
+            if (requestAuthorizationLevel == AuthorizationLevel.Anonymous)
```

Comment:
I do want to break this up into a separate filter, so only the authorization logic is here, but will do that in a separate PR.